visualizations/frontend/filters/v2/TTagsInput.vue

```diff
@@ -0,0 +1,153 @@
+			:loading="loading"
+			:placeholder="placeholder"
+			@opened="handleOpen"
+			@update="handleUpdate"
+			display-key="value"
+			value-key="value"
+		/>
+		<base-tags-input
+			v-else
+			v-model="selectedTags"
+			selection-only
+			:data="tags.length ? tags : tempTags"
+			:loading="loading"
+			:placeholder="placeholder"
+			@opened="handleOpen"
+			@update="handleUpdate"
+		/>
+	</div>
+</template>
+
+<script>
+	export default {
+		props: {
+			label: {
+				type: String,
+				default: '',
+			},
+			tKeyColumn: {
+				type: String,
+				default: 'KEY',
+			},
+			tValueColumn: {
+				type: String,
+				default: 'VALUE',
+			},
+			canReloadOnOpen: {
+				type: Boolean,
+				default: false,
+			},
+			isSingleTag: {
+				type: Boolean,
+				default: false,
+			},
+			placeholder: {
+				type: String,
+				default: '',
+			}
+		},
+		data: () => ({
+			is_filter: true,
+			selectedTags: [],
+			initialized: false,
+			tempTags: [], // base-multiple-value-input does not show selected items if `tags` is empty because data is fetched on open.
+		}),
+		computed:{ 
+			tags() {
+				const items = [];
+
+				if (this.rows.length) {
+					for (let row of this.rows) {
+						const key = row[this.tKeyColumn];
+						const value = row[this.tValueColumn];
+						if (key && value) {
+							items.push({
+								key: key.value,
+								value: value.value,
+							})
+						}
+					}
+				}
+				return items;
+			},
+			firstKey() {
+				// Used for single tag
+				if (this.tags.length) {
+					return this.tags[0].key;
+				}
+				
+				if (this.tempTags) {
+					return this.tempTags[0].key;
+				}
+				
+				return null;
+			}
+		},
+		methods: {
+			onVisualizationInit() {
+				const initial_value = this.getFilterValue("selected_items");
+
+				if (initial_value) {
+					const selectedTags = [];
+                    const tempTags = [];
+					const urlTags = initial_value.split('|');
+					
+					for (let item of urlTags) {
+						const splitPair = item.split('_');
+						const key = splitPair.length > 1 ? splitPair[1] : splitPair[0];
+						const value = splitPair[0];
+
+						if (key && value) {
+							if (this.isSingleTag) {
+								selectedTags.push(value);
+								tempTags.push({ key, value })
+							} else {
+								selectedTags.push({ key, value });
+							}
+						}
+					}
+					
+				    this.tempTags = tempTags;
+				    this.selectedTags = selectedTags;
+				}
+			},
+			handleOpen() {
+				if (!this.initialized || this.canReloadOnOpen) {
+					this.fetchLayerData();
+					this.initialized = true;
+				}
+			},
+			handleUpdate(items) {
+				let urlValue = '';
+				for (let item of items) {
+					if (this.isSingleTag) {
+						if (!this.firstKey) break;
+						let urlString = `${item}_${this.firstKey}|`
+						urlValue += urlString;
+					} else {
+						let urlString = `${item.value}_${item.key}|`
+						urlValue += urlString;
+					}
+				}
+				this.setFilterValue("selected_items", urlValue);
```

Comment:
I'm not quite sure of the format we chose to store tags.
`cve=<value>_<key>|<value>_<key>|<value>_<key>|`
Example:
`cve=ALAS-2018-1010_CVE%257CALAS-2018-1026_CVE%257C`
decoded:
`cve=ALAS-2018-1010_CVE|ALAS-2018-1026_CVE|`

Reply:
Yes this was something that was hard to understand back when we used it for the first time, the reason for this is because of how it is used inside layers.

Layers use filters to compare it like this.

WHERE key = {{ url_filter.key | to_sql_tags  }}
select value

basically `to_sql_tags` expects data in this format inside backend `value_key`, I am sure that backend method can be made better.

Reply:
ok, on that case, I would suggest a small change on the implementation to make this easier. Instead of do some string addition, we can push all the values to an array and then use Array.join method to avoid the trailing `|`.
Example:
```js
const values = [];
for (let item of items) {
	if (this.isSingleTag) {
		if (!this.firstKey) {
		    break;
		}
		values.push(`${item}_${this.firstKey}`);
	} else {
		values.push(`${item.value}_${item.key}`);
	}
}
this.setFilterValue("selected_items", values.join('|'));
```